Framework/Core/src/DataProcessingDevice.cxx

```diff
@@ -77,6 +77,8 @@
 #include <sstream>
 #include <boost/property_tree/json_parser.hpp>
 
+static int g_timeslice_done = -1;
```

Comment:
We should probably make it part of the DataProcessorContext, since "timeslice done" will need to have a different value for each dataprocessor when we move to multiplexing multiple of them to a single device.

Reply:
Yes, but I would think it would be even better if we could, as written above, check in line 1678 when setting the `hasBackpressure = true` what is the timeslice that has backpressure, and then store the minimum timeslice with backpressure.